### ISSUE TITLE
Why grayscaleFilter has been removed ?

### ISSUE THREAD

**Issue by afaucogney** [usefulness/easylauncher-gradle-plugin#74]
First thanks to maintain a fork, that will definitively usefull

Why grayscaleFilter has been removed and how could we replace it ?

**Comment by mateuszkwiecinski** [comment]
Hello @afaucogney 👋 
The history of `grayscaleFilter` is that my primary goal was to add vector icons support, as from my experience most apps use such (related issue: https://github.com/akaita/easylauncher-gradle-plugin/issues/9).
To make that happen I almost completely rewrote the plugin changing base mechanism of how the filters are being applied. (I tried to describe that in the [article I wrote](https://medium.com/@cycki/ribbonize-your-android-application-laucher-icon-again-c52e08d58172))
Previously, the plugin applied customizations directly to your launcher icon, so to make icon greyscale it redraws app icon pixel by pixel removing the color. After library rework, plugin generates layers which are then merged into single `LayerDrawable`. 
That makes it impossible to modify original icon and remove color. 
Moreover, having in mind _Adaptive icons_ support, the customization would have be to applied both to the `foreground` and to the `background` layer, which the plugin doesn't currently support (it only modifies the `foreground`).
I'm not sure I can commit to add support for such modification type. I briefly tried to add such at the very beginning, but heavily failed :/

> how could we replace it?

The easiest solution that comes to my mind is to use adaptive icons and have background layer overriden per each build variant. In the simplest form:
```
<background android:drawable="@color/icon_background_color" />
```
and override `icon_background_color` accroding to your needs.
